internal/search/streaming/search_filters.go

```diff
@@ -107,6 +185,36 @@ func (s *SearchFilters) Update(event SearchEvent) {
 		}
 	}
 
+	addSymbolFilter := func(symbols []*result.SymbolMatch, limitHit bool) {
+		for _, sym := range symbols {
+			selectKind := result.ToSelectKind[strings.ToLower(sym.Symbol.Kind)]
+			filter := fmt.Sprintf(`select:symbol.%s`, selectKind)
+			s.filters.Add(filter, selectKind, 1, limitHit, "symbol type")
+		}
+	}
+
+	addCommitAuthorFilter := func(commit gitdomain.Commit) {
+		author := fmt.Sprintf(`author:%s`, commit.Author.Email)
+		filter := fmt.Sprintf(`type:commit %s`, author)
+		s.filters.Add(filter, commit.Author.Name, 1, false, "author")
+	}
+
+	addCommitDateFilter := func(commit gitdomain.Commit) {
+		var cd time.Time
+		var df DateFilterInfo
+		var filter string
+
+		if commit.Committer != nil {
+			cd = commit.Committer.Date
+		} else {
+			cd = commit.Author.Date
+		}
+
+		df = determineTimeframe(cd)
+		filter = fmt.Sprintf("%s:%s", df.Timeframe, df.Value)
```

Comment:
No need to pre-declare variables. The `:=` operator declares and initializes a variable at the same time, which helps mitigate bugs from using zero-valued variables. 

```suggestion
		var cd time.Time
		if commit.Committer != nil {
			cd = commit.Committer.Date
		} else {
			cd = commit.Author.Date
		}

		df := determineTimeframe(cd)
		filter := fmt.Sprintf("%s:%s", df.Timeframe, df.Value)
```